vortex-duckdb/src/convert/types/to.rs

```diff
@@ -39,7 +40,49 @@ impl ToDuckDBType for DType {
                 );
                 Ok(duckdb_type)
             }
-            DType::List(..) | DType::Extension(_) => todo!(),
+            DType::Extension(ext_dtype) => {
+                if is_temporal_ext_type(ext_dtype.id()) {
+                    Ok(ext_to_duckdb(ext_dtype))
+                } else {
+                    vortex_bail!("Unsupported extension type \"{}\"", ext_dtype.id())
+                }
+            }
+            DType::List(..) => todo!("type: {self:?}"),
+        }
+    }
+}
+
+/// Convert temporal ExtDType to a corresponding LogicalType
+///
+/// panics if the ext_dtype is not a temporal dtype
+pub fn ext_to_duckdb(ext_dtype: &ExtDType) -> LogicalTypeHandle {
+    match TemporalMetadata::try_from(ext_dtype)
+        .vortex_expect("make_arrow_temporal_dtype must be called with a temporal ExtDType")
+    {
+        TemporalMetadata::Date(time_unit) => match time_unit {
+            TimeUnit::D => LogicalTypeHandle::from(LogicalTypeId::Date),
+            _ => {
+                vortex_panic!(InvalidArgument: "Invalid TimeUnit {} for {}", time_unit, ext_dtype.id())
+            }
+        },
+        TemporalMetadata::Time(time_unit) => match time_unit {
+            TimeUnit::Ms => LogicalTypeHandle::from(LogicalTypeId::Date),
+            _ => {
+                vortex_panic!(InvalidArgument: "Invalid TimeUnit {} for {}", time_unit, ext_dtype.id())
+            }
+        },
+        TemporalMetadata::Timestamp(time_unit, tz) => {
+            if tz.is_some() {
+                vortex_panic!(InvalidArgument: "Timestamp with timezone is not yet")
```

Comment:
`Timestamp with timezone is not yet supported`?

Reply:
ye